### PR TITLE
Added more functions to emulate numpy ndarray

### PR DESCRIPTION
**Pull request by kirit93**
This branch adds more functionality the FixedPointArray. Now the class is more similar to a Numpy array and can support most numpy array functions.

### CODE REVIEW

**Comment by hgomersall** [comment]
This is valuable work, and very much appreciated, but I would entreat you to flesh out tests to exercise this - it _should_ be a relatively easy task for each one (based on the other tests in `tests`, but I picked up loads of corner cases when I did this myself).

**Comment by kirit93** [comment]
Sure thing. It may take me sometime to add all the tests since I'm working on a couple of projects at work right now. However I will add the tests as soon as possible. Let me know if there are any other functions you think might be useful that I missed out.

File: fixpy/fixed_point.py [hunk 1]
```diff
@@ -124,29 +114,48 @@ def has_negative_part(array):
...
         return max_bits
 
-
     def __getitem__(self, _slice):
-
         sliced_data = self.data[_slice]
-        return self.__class__(sliced_data,
-                fractional_bits=self.fractional_bits,
-                data_already_scaled=True)
+        return self.__class__(sliced_data, fractional_bits=self.fractional_bits, data_already_scaled=True)
```
Comment (hgomersall) [inline fixpy/fixed_point.py]: Quick FYI, the long lines break [pep8](https://www.python.org/dev/peps/pep-0008/#maximum-line-length). Not a big deal, but a nice to have.

**Comment by hgomersall** [comment]
Great! No rush (it doesn't seem to a project that is moving quickly!), but it's nice to get it right.

I suggest for the time being just stick with what you've done, unless you're keen to do more. My philosophy on these sorts of things is [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it), which is why I hadn't done it already.
